dev/src/filter.ts

```diff
@@ -0,0 +1,222 @@
+ * be used to refine the results of a {@link Query}.
+ * `Filters`s are created by invoking {@link Filter#where}, {@link Filter#or},
+ * or {@link Filter#and} and can then be passed to {@link Query#where}
+ * to create a new {@link Query} instance that also contains this `Filter`.
+ */
+export abstract class Filter {
+  /**
+   * Creates and returns a new [Filter]{@link Filter}, which can be
+   * applied to [Query.where()]{@link Query#where}, [Filter.or()]{@link Filter#or},
+   * or [Filter.and()]{@link Filter#and}. When applied to a [Query]{@link Query}
+   * it requires that documents must contain the specified field and that its value should
+   * satisfy the relation constraint provided.
+   *
+   * @param {string|FieldPath} fieldPath The name of a property value to compare.
+   * @param {string} opStr A comparison operation in the form of a string.
+   * Acceptable operator strings are "<", "<=", "==", "!=", ">=", ">", "array-contains",
+   * "in", "not-in", and "array-contains-any".
+   * @param {*} value The value to which to compare the field for inclusion in
+   * a query.
+   * @returns {Filter} The created Filter.
+   *
+   * @example
+   * ```
+   * let collectionRef = firestore.collection('col');
+   *
+   * collectionRef.where(Filter.where('foo', '==', 'bar')).get().then(querySnapshot => {
+   *   querySnapshot.forEach(documentSnapshot => {
+   *     console.log(`Found document at ${documentSnapshot.ref.path}`);
+   *   });
+   * });
+   * ```
+   */
+  public static where(
+    fieldPath: string | firestore.FieldPath,
+    opStr: firestore.WhereFilterOp,
+    value: unknown
+  ): Filter {
+    return new UnaryFilter(fieldPath, opStr, value);
+  }
+
+  /**
+   * Creates and returns a new [Filter]{@link Filter} that is a
+   * disjunction of the given {@link Filter}s. A disjunction filter includes
+   * a document if it satisfies any of the given {@link Filter}s.
+   *
+   * The returned Filter can be applied to [Query.where()]{@link Query#where},
+   * [Filter.or()]{@link Filter#or}, or [Filter.and()]{@link Filter#and}. When
+   * applied to a [Query]{@link Query} it requires that documents must satisfy
+   * one of the provided {@link Filter}s.
+   *
+   * @param {...Filter} filters  Optional. The {@link Filter}s
+   * for OR operation. These must be created with calls to {@link Filter#where},
+   * {@link Filter#or}, or {@link Filter#and}.
+   * @returns {Filter} The created {@link Filter}.
+   *
+   * @example
+   * ```
+   * let collectionRef = firestore.collection('col');
+   *
+   * // doc.foo == 'bar' || doc.baz > 0
+   * let orFilter = Filter.or(Filter.where('foo', '==', 'bar'), Filter.where('baz', '>', 0));
+   *
+   * collectionRef.where(orFilter).get().then(querySnapshot => {
+   *   querySnapshot.forEach(documentSnapshot => {
+   *     console.log(`Found document at ${documentSnapshot.ref.path}`);
+   *   });
+   * });
+   * ```
+   */
+  public static or(...filters: Filter[]): Filter {
```

Comment:
This came up in API design review and we decided to go with this approach because it is more consistent with existing static functions in the nodejs-firestore SDK. See for example, the [FieldValue class](https://googleapis.dev/nodejs/firestore/latest/FieldValue.html).